include/openrave/planner.h

```diff
@@ -78,7 +80,9 @@ enum PlannerAction
 enum NeighborStateOptions
 {
     NSO_GoalToInitial=1, ///< if set, then q is coming from a goal state, else it is coming from an initial state.
-    NSO_OnlyHardConstraints=2, ///< if set, then the new neighbor should be as close as possible to q+qdelta, otherwise can prioritize other constraints and only use q+qdelta as a hint. This is used in smoothers when the path is already determined and user just wants to verify that hard constraints are met only; do not modify q+qdelta unless hard constraints fail.
+    NSO_OnlyHardConstraints=2, ///< if set, then the new neighbor should be as close as possible to q+qdelta. This is used in smoothers when the path is already determined and user just wants to verify that hard constraints are met only; do not modify q+qdelta unless hard constraints fail.
+    NSO_TighterConstraintsForShortcutting=4, ///< if set, then the new neighbor should be as close as possible to q+qdelta. the new neighbor needs to preserve tighter constraint than hard constraint to give a room for the later processing. This is used in shortcutter when the path is already determined and user just wants to verify that tighter constraints are met only; do not modify q+qdelta unless tighter constraints fail.
```

Comment:
@yoshikikanemoto I guess the purpose here is to have the neighstate function internally decide how tight the constraints should be. But maybe in the future, we would want different behaviors depending on this. In that case, perhaps renaming the enums could be `NSO_FromPathSampling` (ie birrt), `NSO_FromPathShortcutting` (ie linear shortcut), `NSO_FromTrajectorySmoother` (ie parabolic smoohter)

Reply:
yes. i want to change behavior of neighstate function depending on from where it is called. you suggestion of changing the enum is clearer because it differentiates shortcut and smoother. i actually felt a little bit inappropriate not giving anything to smoother since smoother can be called some sort of shortcutting algorithm. i'll change it accordingly